### PR TITLE
Curating Owners: pkg/client

### PR DESCRIPTION
**Pull request by apelisse**
cc @lavalamp @caesarxuchao @mikedanese @timothysc @smarterclayton @krousey @deads2k

In an effort to expand the existing pool of reviewers and establish a
two-tiered review process (first someone lgtms and then someone
experienced in the project approves), we are adding new reviewers to
existing owners files.

If You Care About the Process:
------------------------------

We did this by algorithmically figuring out who’s contributed code to
the project and in what directories.  Unfortunately, that doesn’t work
well: people that have made mechanical code changes (e.g change the
copyright header across all directories) end up as reviewers in lots of
places.

Instead of using pure commit data, we generated an excessively large
list of reviewers and pruned based on all time commit data, recent
commit data and review data (number of PRs commented on).

At this point we have a decent list of reviewers, but it needs one last
pass for fine tuning.

Also, see https://github.com/kubernetes/contrib/issues/1389.

TLDR:
-----

As an owner of a sig/directory and a leader of the project, here’s what
we need from you:

1. Use PR https://github.com/kubernetes/kubernetes/pull/35715 as an example.

2. The pull-request is made editable, please edit the `OWNERS` file to
remove the names of people that shouldn't be reviewing code in the
future in the **reviewers** section. You probably do NOT need to modify
the **approvers** section. Names asre sorted by relevance, using some
secret statistics.

3. Notify me if you want some OWNERS file to be removed.  Being an
approver or reviewer of a parent directory makes you a reviewer/approver
of the subdirectories too, so not all OWNERS files may be necessary.

4. Please use ALIAS if you want to use the same list of people over and
over again (don't hesitate to ask me for help, or use the pull-request
above as an example)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36526)

### CODE REVIEW

File: pkg/client/restclient/OWNERS [hunk 1]
```diff
@@ -0,0 +1,24 @@
+reviewers:
```
Comment (caesarxuchao) [inline pkg/client/restclient/OWNERS]: @apelisse could you add me to the reviewers of this file? Seems the files are not editable.

Comment (apelisse) [inline pkg/client/restclient/OWNERS]: @deads2k @caesarxuchao sorry I had to make the PR editable manually. I've updated the gazillion pull-request to make them editable. You should be good to go now.

Comment (caesarxuchao) [inline pkg/client/restclient/OWNERS]: I'm also the reviewer of the top-level folder, so I don't need to be reviewer of this subdirectory.

(@deads2k I talked with @apelisse offline. He said being a reviewer means one would review the details of PR, while being an approver means one would review the intent of the PR, so being an approver is not a superset of being an reviewer)

**Comment by apelisse** [comment]
his name is missing from the list, but consider adding "AdoHe"

File: pkg/client/restclient/OWNERS (hunk 1 above)
Comment (lavalamp) [inline pkg/client/restclient/OWNERS]: I added Chao to the list.

**Comment by lavalamp** [comment]
I made a quick pass.

**Comment by apelisse** [comment]
Squashed @lavalamp, can we get this merged now? Please lgtm @mikedanese

**Comment by caesarxuchao** [comment]
lgtm. lavalamp is OOO, I saw his commit was incorporated, so I'll "lgtm" on his behalf as well.

**Pull request merged**